### PR TITLE
fixes for preview update

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**

### CODE REVIEW

File: src/app/modules/map-config/components/preview/preview.component.ts [hunk 1]
```diff
@@ -187,70 +188,58 @@ export class PreviewComponent implements AfterViewInit, OnDestroy {
...
+    const resourcesConfig = this.mainFormService.resourcesConfig.getFg();
+    resourcesConfig.customControls.resources.previewValue.setValue(img);
+    console.log(previewExists);
+    if (previewExists) {
+      this.persistenceService.updatePreview(previewId, previewReaders, previewWriters, img);
+    } else {
+      console.log('yey');
+      this.persistenceService.create(ZONE_PREVIEW, name, img, previewReaders, previewWriters)
+        .pipe(map((p: DataWithLinks) => {
+          console.log('creating');
```
Comment (mbarbet) [inline src/app/modules/map-config/components/preview/preview.component.ts:230]: remove all console.log

**Review by mbarbet — APPROVED** [review]

File: src/app/services/resources-form-builder/resources-config-form-builder.service.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      previewValue: this.get('resources.previewValue'),
+    }
+  };
+
+  public hasPreviewId() {
+    return !!this.customControls.resources.previewId.value && this.customControls.resources.previewId.value !== '';
+  }
+
+  public hasPreviewImage() {
+    return !!this.customControls.resources.previewValue.value && this.customControls.resources.previewValue.value !== '';
```
Comment (QuCMGisaia) [inline src/app/services/resources-form-builder/resources-config-form-builder.service.ts:46]: !! already checks if the string is empty

File: src/settings.yaml [hunk 3]
```diff
@@ -20,20 +20,16 @@ authentication:
...
+  logout_url: https://arlas-cloud-dev.eu.auth0.com/v2/logout?returnTo=http://localhost:4200
   storage: sessionstorage
   custom_query_params: [{"audience":"http://arlas.io/api/server"}]
-
-########################################################
-############   Persistence SETTINGS   ##################
-########################################################
 persistence:
-  url: 'http://localhost:19997/arlas-persistence-server'
+  url: 'https://demo.cloud.staging.arlas.io/arlas/persistence'
```
Comment (QuCMGisaia) [inline src/settings.yaml:27]: Do we want to put those urls in the toolkit ?

**Review by QuCMGisaia — APPROVED** [review]

File: src/app/services/main-form-manager/main-form-manager.service.ts [hunk 4]
```diff
@@ -272,36 +253,47 @@ export class MainFormManagerService {
     this.updateControlsFromOtherControls(this.mainFormService.mainForm);
   }
 
+  /**
+   * The preview should be created if the a preview image has been captured by the user but no identifier has yet been created.
+   * @param startForm The form where the preview image is stored
+   * @returns Whether we should create the preview in persistence or not
+   */
+  private shoulCreatePreview(startForm: ResourcesConfigFormGroup) {
```
[the remark this answers is not in the document]
Comment (MohamedHamouGisaia) [reply src/app/services/main-form-manager/main-form-manager.service.ts:261]: fixed

File: src/app/services/resources-form-builder/resources-config-form-builder.service.ts (hunk 2 above)
Comment (MohamedHamouGisaia) [reply src/app/services/resources-form-builder/resources-config-form-builder.service.ts]: fixed

**Pull request merged**